modules/mono/glue/GodotSharp/GodotSharp/Core/StringExtensions.cs

```diff
@@ -728,6 +728,26 @@ public static uint Hash(this string instance)
             return hash;
         }
 
+        /// <summary>
+        /// Decodes a hexadecimal string.
+        /// </summary>
+        /// <param name="instance">The hexadecimal string.</param>
+        /// <returns>The byte array representation of this string.</returns>
+        public static byte[] HexDecode(this string instance)
+        {
+            if (instance.Length % 2 != 0)
+            {
+                throw new ArgumentException("Hexadecimal string of uneven length.", nameof(instance));
+            }
+            int len = instance.Length / 2;
+            byte[] ret = new byte[len];
+            for (int i = 0; i < len; i++)
+            {
+                ret[i] = (byte)int.Parse(instance.Substring(i * 2, 2), NumberStyles.HexNumber);
```

Comment:
```suggestion
                ret[i] = (byte)int.Parse(instance.AsSpan(i * 2, 2), NumberStyles.AllowHexSpecifier);
```
cc @raulsntos I think we can use spans? This would avoid unneeded allocations.

Also note that [`NumberStyles.HexNumber`](https://learn.microsoft.com/en-gb/dotnet/api/system.globalization.numberstyles?view=net-7.0#system-globalization-numberstyles-hexnumber) allows leading/trailing whitespaces, which means this method wouldn't fail e.g. for input like "1 1" (the space inbetween would be ignored as a trailing one as at most 2 chars are parsed at once), contrary to the core method in `ustring.cpp` which would error for such input. I doubt such discrepancy is wanted and hence it likely makes more sense to pass [`NumberStyles.AllowHexSpecifier`](https://learn.microsoft.com/en-gb/dotnet/api/system.globalization.numberstyles?view=net-7.0#system-globalization-numberstyles-allowhexspecifier) in here instead?

Reply:
How does `AsSpan` work?

Reply:
> How does `AsSpan` work?

[MemoryExtensions.AsSpan(String, Int32, Int32)](https://learn.microsoft.com/en-gb/dotnet/api/system.memoryextensions.asspan?view=net-7.0#system-memoryextensions-asspan(system-string-system-int32-system-int32))

Reply:
> I think we can use spans? This would avoid unneeded allocations.

Sure, I think there are a lot of places in `StringExtensions.cs` that could use `Span` but I didn't want to burden a first time contributor with it, since I don't think it's necessary for the initial implementation.

> Also note that [`NumberStyles.HexNumber`](https://learn.microsoft.com/en-gb/dotnet/api/system.globalization.numberstyles?view=net-7.0#system-globalization-numberstyles-hexnumber) allows leading/trailing whitespaces [...] I doubt such discrepancy is wanted and hence it likely makes more sense to pass [`NumberStyles.AllowHexSpecifier`](https://learn.microsoft.com/en-gb/dotnet/api/system.globalization.numberstyles?view=net-7.0#system-globalization-numberstyles-allowhexspecifier) in here instead?

I think it makes sense for the implementation to match Core behavior, if that's `NumberStyles.AllowHexSpecifier` then maybe we should change it in `HexToInt` too (although this would break compatibility, so on second thought maybe not).

https://github.com/godotengine/godot/blob/f1521a55cf872f76dbfa417a70bd24f3c33d804c/modules/mono/glue/GodotSharp/GodotSharp/Core/StringExtensions.cs#L825

However, note that C# implementation sometimes differs from Core to try to be more _idiomatic_. We may prefer to be more forgiving here and allow trailing and leading white-space but this could be added later without breaking compatibility.